src/k4a_ros_device.cpp

```diff
@@ -86,9 +86,17 @@ K4AROSDevice::K4AROSDevice(const NodeHandle &n, const NodeHandle &p) : k4a_devic
             params_.rgb_point_cloud = false;
         }
         // This is necessary because at the moment there are only checks in place which use BgraPixel size
-        else if (params_.color_enabled && record_config.color_track_enabled && record_config.color_format != K4A_IMAGE_FORMAT_COLOR_BGRA32)
+        else if (params_.color_enabled && record_config.color_track_enabled)
         {
-            k4a_playback_handle_.set_color_conversion(K4A_IMAGE_FORMAT_COLOR_BGRA32);
+            if (params_.color_format == "jpeg" && record_config.color_format != K4A_IMAGE_FORMAT_COLOR_MJPG)
+            {
+                ROS_WARN("Converting color images to K4A_IMAGE_FORMAT_COLOR_MJPG is not supported.");
```

Comment:
I think I'd rather have the node throw an exception here: I'd consider this an invalid parameter passed in on the command line, and the general way we've been treating invalid parameters is to exit with an error.

Reply:
Done